IPython/core/debugger.py

```diff
@@ -329,8 +332,25 @@ def print_stack_trace(self, context=None):
         except (TypeError, ValueError):
                 raise ValueError("Context must be a positive integer")
         try:
-            for frame_lineno in self.stack:
+            ip_hide = [s[0].f_locals.get("__tracebackhide__", False) for s in self.stack]
+            ip_start = [i for i,s in enumerate(ip_hide) if s == '__ipython_bottom__']
```

Comment:
This is a special value in IPython, otherwise it is impossible to hide all the stacks we want. Typically: 

```
ipdb> w
  /Users/bussonniermatthias/miniconda3/bin/ipython(11)<module>()
      8 if __name__ == '__main__':
      9     sys.argv[0] = re.sub(r'(-script\.pyw?|\.exe)?$', '', sys.argv[0])
     10     sys.exit(
---> 11         load_entry_point('ipython', 'console_scripts', 'ipython')()
     12     )
```

I use this special value to say "everything above, ignore". Otherwise yes it is boolean.